test/export/StructureDefinitionExporter.test.ts

```diff
@@ -2294,6 +2410,620 @@ describe('StructureDefinitionExporter', () => {
+      expect(sd.findElement('Observation.component:Lab.interpretation').max).toBe('9');
+      expect(loggerSpy.getLastMessage('error')).toMatch(/cannot be narrowed/s);
+      expect(loggerSpy.getLastMessage('error')).toMatch(/File: Narrower\.fsh.*Line: 7\D*/s);
+    });
+
+    it('should apply a FlagRule on a sliced element that updates the flags on its slices', () => {
+      // * component MS
+      const rootFlag = new FlagRule('component');
+      rootFlag.mustSupport = true;
+
+      observationWithSlice.rules.push(rootFlag);
+      doc.profiles.set(observationWithSlice.name, observationWithSlice);
+      exporter.export();
+      const sd = pkg.profiles[0];
+      const rootComponent = sd.findElement('Observation.component');
+      const labComponent = sd.findElement('Observation.component:Lab');
+      expect(rootComponent.mustSupport).toBe(true);
+      expect(labComponent.mustSupport).toBe(true);
+    });
+
+    it('should apply a FlagRule on a sliced element that disables a no-disable flag, but not disable the flag on the slice', () => {
+      // * component[Lab] MS
+      // * component ~MS // currently, there is no FSH to explicitly set a flag to false
+      const labFlag = new FlagRule('component[Lab]');
+      labFlag.mustSupport = true;
+      const rootFlag = new FlagRule('component');
+      rootFlag.mustSupport = false;
+
+      observationWithSlice.rules.push(labFlag, rootFlag);
+      doc.profiles.set(observationWithSlice.name, observationWithSlice);
+      exporter.export();
+      const sd = pkg.profiles[0];
+      const rootComponent = sd.findElement('Observation.component');
+      const labComponent = sd.findElement('Observation.component:Lab');
+      expect(rootComponent.mustSupport).toBe(false);
+      expect(labComponent.mustSupport).toBe(true);
+    });
+
+    it('should apply a FlagRule on the child of a sliced element that updates the flags on the child of a slice', () => {
+      // * component[Lab].interpretation 0..1 // this forces the creation of the unfolded slice
+      // * component.interpretation MS
+      const labCard = new CardRule('component[Lab].interpretation');
+      labCard.min = 0;
+      labCard.max = '1';
+      const rootFlag = new FlagRule('component.interpretation');
+      rootFlag.mustSupport = true;
+
+      observationWithSlice.rules.push(labCard, rootFlag);
+      doc.profiles.set(observationWithSlice.name, observationWithSlice);
+      exporter.export();
+      const sd = pkg.profiles[0];
+      const rootInterpretation = sd.findElement('Observation.component.interpretation');
+      const labInterpretation = sd.findElement('Observation.component:Lab.interpretation');
+      expect(rootInterpretation.mustSupport).toBe(true);
+      expect(labInterpretation.mustSupport).toBe(true);
+    });
+
+    it('should not apply a flag rule that would disable a must support or is modifier flag on a slice when the list has that flag', () => {
```

Comment:
Ok so I see the different behavior of `isSummary` is definitely on purpose, but why?

Reply:
The justification is this:
* disabling `mustSupport` changes the software requirements of systems handling this resource, which means that you could end up with conflicting requirements (the parent resource says you have to support something, but the child resource says you don't.)
* disabling `modifier` changes the meaning of an element, which in my opinion is a way to make things more confusing.
* disabling `summary` changes whether an element appears in the resource's summary. The element still exists, and means the same thing, and still carries the same software requirements. If you need the element, you can directly ask for it. So, disabling `summary` seems less likely to cause issues.

Of course, all of this is still 1. up for discussion and 2. maybe meaningless, since there's no way in FSH to explicitly disable flags.

Reply:
See also
* http://hl7.org/fhir/R4/elementdefinition-definitions.html#ElementDefinition.mustSupport 
* http://hl7.org/fhir/R4/elementdefinition-definitions.html#ElementDefinition.isModifier
* http://hl7.org/fhir/R4/elementdefinition-definitions.html#ElementDefinition.isSummary

Although, looking more closely at `isModifier`, it looks like there are even more detailed rules going on here that aren't being implemented correctly: `Only the definition of an element can set IsModifier true - either the specification itself or where an extension is originally defined. Once set, it cannot be changed in derived profiles.`

Reply:
Ok I buy that reasoning for `summary`, and good catch on `isModifier`, but that could be a separate task I think.